packages/@aws-cdk/custom-resource-handlers/lib/aws-eks/cluster-resource-handler/fargate.ts

```diff
@@ -34,9 +34,15 @@ export class FargateProfileResourceHandler extends ResourceHandler {
       throw new Error('Cannot delete a profile without a physical id');
     }
 
+    // Use the same logic as onCreate to determine the profile name
+    // If fargateProfileName was explicitly provided, use it; otherwise use physicalResourceId
```

Comment:
Based on the logic in `onCreate` (line 25), `physicalResourceId` is the same as the `fargateProfileName`

```
PhysicalResourceId: createFargateProfileResponse.fargateProfile.fargateProfileName,
```